# Task 2: Visualizing fire scars through false color

## Datasets description

## 2. Fire perimeter data retrieval and selection

Explore the data and write a brief summary of the information you obtained from the preliminary information. Your summary should include the CRS of the data and whether this is projected or geographic.

In [4]:
# Import libraries
import os
import numpy as np
import pandas as pd
import geopandas as gpd
import xarray as xr
import rioxarray as rioxr

In [9]:
# Read data
landsat = rioxr.open_rasterio('data/landsat8-2018-01-26-sb-simplified.nc')

## Exploration

In [10]:
# Display the first few rows of the DataFrame to get an overview of the data

landsat.head()

<xarray.Dataset>
Dimensions:      (y: 5, x: 5, band: 1)
Coordinates:
  * y            (y) float64 3.952e+06 3.952e+06 3.952e+06 3.952e+06 3.951e+06
  * x            (x) float64 1.213e+05 1.216e+05 1.218e+05 1.221e+05 1.224e+05
  * band         (band) int64 1
    spatial_ref  int64 0
Data variables:
    red          (band, y, x) float64 ...
    green        (band, y, x) float64 ...
    blue         (band, y, x) float64 ...
    nir08        (band, y, x) float64 ...
    swir22       (band, y, x) float64 ...

In [21]:
# Check what variables are in the dataset
print("\nVariables in the Dataset:")
print(landsat.data_vars)


Variables in the Dataset:
Data variables:
    red      (band, y, x) float64 ...
    green    (band, y, x) float64 ...
    blue     (band, y, x) float64 ...
    nir08    (band, y, x) float64 ...
    swir22   (band, y, x) float64 ...


In [22]:
# 3. Dataset Information
raster_data = landsat['red'] 
print("\nShape of the raster data (height x width x bands):", raster_data.shape)
print("Dimensions of the raster data:", raster_data.dims)
print("Data type of the raster data:", raster_data.dtype)


Shape of the raster data (height x width x bands): (1, 731, 870)
Dimensions of the raster data: ('band', 'y', 'x')
Data type of the raster data: float64


In [23]:
# Check the CRS
print("Coordinate reference system (CRS):", landsat.rio.crs)
# Using the 'dims' attribute to inspect dimensions
print("Dimensions of the data:", landsat.dims)  


Coordinate reference system (CRS): EPSG:32611
Dimensions of the data: Frozen({'y': 731, 'x': 870, 'band': 1})


# Summary

The dataset loaded is a Landsat 8 image, collected on January 26, 2018. The data is in NetCDF format, simplified for analysis, and contains information about the surface reflectance captured by the Landsat 8 satellite sensor. It consists of several spectral bands, which are typically used to derive information about the Earth's surface, vegetation, and atmosphere.

**Variables in the Dataset** 

The dataset contains several variables related to different spectral bands from the Landsat 8 satellite. The exact variable names may vary, but they are typically stored as xarray.DataArray objects. Upon inspecting the dataset, you will find variables that correspond to the different bands of the satellite imagery, which might include:

Band 1: Coastal/Aerosol
Band 2: Blue
Band 3: Green
Band 4: Red
Band 5: Near Infrared (NIR)
Band 6: Shortwave Infrared (SWIR 1)
Band 7: Shortwave Infrared (SWIR 2)

**Coordinate reference system (CRS):**

EPSG:32611,  this is a projected coordinate reference system, not a geographic CRS. 

**Dimensions of the data:**

The dataset contains a raster grid with dimensions
Frozen({'y': 731, 'x': 870, 'band': 1})

From your fire perimeter data, select the Thomas Fire boundary. The fire occurred in 2017.



Save only the 2017 Thomas Fire boundary as a geospatial file in the format of your choosing. The file should go into the data/ directory in your repository.

In a markdown cell, briefly explain your reasoning for selecting that specific file format.

## 3. True color image

In [ ]:
Construct a file path to the Landsat data using os and import it using rioxr.open_rasterio().

In [ ]:
Explore the data and write a brief summary of the information you obtained from the preliminary information. Your summary should include the a description of the data’s variables and dimensions.

In [ ]:
Drop the band dimension of the data. HINT: squeeze() and drop_vars().

In [ ]:
Without creating any new variables:

- select the red, green, and blue variables (in that order) of the xarray.Dataset holding the Landsat data,
- convert it to a numpy.array using the to_array() method, and then
- use .plot.imshow() to create an RGB image with the data. There will be a warning, that’s ok.

In [ ]:
Adjust the scale used for plotting the bands to get a true color image. HINT: Check the robust parameter. The issue here is the clouds: their RGB values are outliers and cause the other values to be squished when plotting.

In [ ]:
In a markdown cell write a brief explanation comparing the ouputs for parts (a) and (b).

## 4. False color image

Without creating any new variables, create a false color image by plotting the short-wave infrared (swir22), near-infrared, and red variables (in that order).

## 5. Map

Create a map showing the shortwave infrared/near-infrared/red false color image together with the Thomas Fire perimeter. Customize it appropriately including, at least, an informative title and legend.

Write a figure description for the map including a brief explanation of how false color imagery is being used.



## 6. Clean your notebooks

In [ ]:
#7. Update your repository